packages/open-payments/src/client/grant.ts

```diff
@@ -53,11 +56,24 @@ export const createGrantRoutes = (deps: GrantRouteDeps): GrantRoutes => {
   }
 
   return {
-    request: (
+    request: async (
       { url }: UnauthenticatedResourceRequestArgs,
       args: Omit<GrantRequest, 'client'>
-    ) =>
-      post(
+    ) => {
+      const outgoingPaymentAccess = args.access_token.access.find(
+        (el) => el.type === 'outgoing-payment'
+      )
+      if (
+        (outgoingPaymentAccess?.limits as OutgoingPaymentLimitWithDebitAmount)
+          ?.debitAmount &&
+        (outgoingPaymentAccess?.limits as OutgoingPaymentLimitWithReceiveAmount)
+          ?.receiveAmount
```

Comment:
_(if we proceed with the new types)_
```suggestion
        (outgoingPaymentAccess?.limits as OutgoingAccessWithDebitAmount)
          ?.debitAmount &&
        (outgoingPaymentAccess?.limits as OutgoingAccessWithReceiveAmount)
          ?.receiveAmount
```

Reply:
This looks great! Thanks for writing it up. The discussion on native XOR implementation is _very_ spirited...